pkg/retry/retry.go

```diff
@@ -0,0 +1,63 @@
+// package retry has utilities to retry operations
+package retry
+
+import (
+	"time"
+)
+
+type config struct {
+	interval    time.Duration
+	maxAttempts int
+}
+
+// Option allows to configure the behavior of retry.Do
+type Option func(*config)
```

Comment:
Do we need to export this type?

Reply:
I think yes, for two reasons:

- my understanding is that generally is good to export any type your functions are returning
- all other modules in the codebase using this pattern (`Options` + `With...`) have this exported, so at least I'd say we should export it for consistency.